components/gitlab/new-project.js

```diff
@@ -0,0 +1,89 @@
+const gitlab = require("https://github.com/PipedreamHQ/pipedream/components/gitlab/gitlab.app.js");
+
+module.exports = {
+  name: "New Project",
+  description: "Triggers when a new project (i.e. repository) is created",
+  version: "0.0.1",
+  dedupe: "greatest",
+  props: {
+    gitlab,
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 15 * 60, // 15 minutes
+      },
+    },
+    http: "$.interface.http",
+    db: "$.service.db",
+  },
+  hooks: {
+    async activate() {
```

Comment:
Hm, I think I like this pattern of using the activate hook for this. We should backport it to older components, although in older components we typically do emit a recent page (eg. for test event reasons). We should tackle that issue differently though -- this is good.

Reply:
+1
We should definitely strive for consistent building blocks across the project.